libp2p/muxers/muxer.nim

```diff
@@ -46,11 +46,14 @@ proc newMuxerProvider*(creator: MuxerConstructor, codec: string): MuxerProvider
 method init(c: MuxerProvider) =
   proc handler(conn: Connection, proto: string) {.async, gcsafe, closure.} =
     let muxer = c.newMuxer(conn)
+    var handlerFut: Future[void] = newFuture[void]()
```

Comment:
```
let handlerFut =
  if ..: c.muxerHandler(..)
  else: dummy future
```

`let` is a safer construct (even like that, the code is wasteful when always calling `allFutures` - it also smells with `streamHandler` being set midway after `muxerHandler` is caled and `isNil` checked twice etc, classic hotspot for bugs)

Reply:
Thanks! Yes, this is just me moving things around trying to figure out where the raise is, this wasn't part of the original PR.